### ISSUE TITLE
Can't write compiler with type "misc"

### ISSUE THREAD

**Issue by zaceno** [dbashford/mimosa#391]
tl; dr: Either config.extensions should have a "misc" key on initialization, or I am approaching my problem wrong (- and would appreciate some advice, thanks :) )

I am trying to write a compiler-module whose type is not naturally js, css, template or copy. So I settled on "misc".

The situation is that I have a legacy project tree with .license files spread out, which need to be read & fed through a processor and templating to produce a single LICENSE.txt in the build. Since I want this license-file-building to behave the same as when I edit a js file or sass file, I figured a compiler module was appropriate.

Problem is: mimosa throws an error at the point of registering my compiler's extensions. Mimosa tries to push my extensions to config.extensions["misc"], but config.extensions doesn't come with a "misc" key.

In general, I can see plenty of use cases where you might want to write a specialized compiler for things that don't fit neatly into any of the apparent types mimosa handles (css, js, copy, misc, template). I assume most such cases should go into misc.

A brief explanation of how mimosa handles compilers of different types would be interesting! (i e: why the differentiation?)
